src/core/imageUtils.js

```diff
@@ -0,0 +1,10 @@
+/* global navigator, window */
+/* eslint-disable react/prop-types */
+
+import { isAllowedOrigin } from 'core/utils';
+import fallbackIcon from 'amo/img/icons/default-64.png';
+
+export function getIconUrl(addon) {
+  return (addon && isAllowedOrigin(addon.icon_url)) ? addon.icon_url : fallbackIcon;
```

Comment:
This should have a line above it checking that `addon` is not empty, like:

```js
if (!addon) {
  throw new Error('addon cannot be empty');
}
```
This will also need a unit test